### PR TITLE
Run pyupgrade on codebase

### PR DESCRIPTION
**Pull request by thedrow**
*Note*: Before submitting this pull request, please review our [contributing
guidelines](https://docs.celeryproject.org/en/master/contributing.html).

## Description

pyupgrade is an automatic tool that upgrades codebases to Python 3.
I think it did the trick pretty well.
The unit tests pass so that's something.

### CODE REVIEW

**Comment by auvipy** [comment]
looks really interesting!!!

File: t/unit/worker/test_worker.py [hunk 1]
```diff
@@ -41,7 +41,7 @@ def MockStep(step=None):
     else:
         step.blueprint = Mock(name='step.blueprint')
     step.blueprint.name = 'MockNS'
-    step.name = 'MockStep(%s)' % (id(step),)
+    step.name = 'MockStep({})'.format(id(step))
```
Comment (auvipy) [inline t/unit/worker/test_worker.py]: no f string here?

File: t/unit/tasks/test_result.py [hunk 2]
```diff
@@ -294,13 +294,13 @@ def test_repr(self):
         ok_res = self.app.AsyncResult(self.task1['id'])
         ok2_res = self.app.AsyncResult(self.task2['id'])
         nok_res = self.app.AsyncResult(self.task3['id'])
-        assert repr(ok_res) == '<AsyncResult: %s>' % (self.task1['id'],)
-        assert repr(ok2_res) == '<AsyncResult: %s>' % (self.task2['id'],)
-        assert repr(nok_res) == '<AsyncResult: %s>' % (self.task3['id'],)
+        assert repr(ok_res) == '<AsyncResult: {}>'.format(self.task1['id'])
```
Comment (auvipy) [inline t/unit/tasks/test_result.py]: what about using f string here as well?

File: t/unit/conftest.py [hunk 3]
```diff
@@ -319,6 +319,6 @@ def import_all_modules(name=__name__, file=__file__,
                 pass
             except OSError as exc:
                 warnings.warn(UserWarning(
-                    'Ignored error importing module {0}: {1!r}'.format(
+                    'Ignored error importing module {}: {!r}'.format(
```
Comment (auvipy) [inline t/unit/conftest.py]: is this alright?

File: t/unit/backends/test_cache.py [hunk 4]
```diff
@@ -156,10 +156,10 @@ def set(self, key, value, *args, **kwargs):
             key_t, must_be, not_be, cod = text_t, 'bytes', 'string', 'encode'
         if isinstance(key, key_t):
             raise MyMemcachedStringEncodingError(
-                'Keys must be {0}, not {1}.  Convert your '
-                'strings using mystring.{2}(charset)!'.format(
+                'Keys must be {}, not {}.  Convert your '
```
Comment (auvipy) [inline t/unit/backends/test_cache.py]: is this format alright? can we also use f'' here?

File: t/unit/apps/test_multi.py [hunk 5]
```diff
@@ -123,10 +123,10 @@ def _args(name, *args):
             'COMMAND', '-c 5', '-n celery1@example.com') + _args('celery1')
         for i, worker in enumerate(nodes3[1:]):
             assert worker.name == 'celery%s@example.com' % (i + 2)
-            node_i = 'celery%s' % (i + 2,)
+            node_i = 'celery{}'.format(i + 2)
```
Comment (auvipy) [inline t/unit/apps/test_multi.py]: f' ' instead of  .format?

File: t/unit/app/test_schedules.py [hunk 6]
```diff
@@ -79,7 +79,7 @@ def test_event_uses_center(self):
             try:
                 s.remaining_estimate(datetime.utcnow())
             except TypeError:
-                pytest.fail("{0} was called with 'use_center' which is not a \
+                pytest.fail("{} was called with 'use_center' which is not a \
```
Comment (auvipy) [inline t/unit/app/test_schedules.py]: can you manually check this conversion?

File: t/unit/app/test_control.py [hunk 7]
```diff
@@ -44,7 +44,7 @@ def test_flatten_reply(self):
         with pytest.warns(DuplicateNodenameWarning) as w:
             nodes = control.flatten_reply(reply)
 
-        assert 'Received multiple replies from node name: {0}.'.format(
+        assert 'Received multiple replies from node name: {}.'.format(
```
Comment (auvipy) [inline t/unit/app/test_control.py]: f string?

File: t/distro/test_CI_reqs.py [hunk 8]
```diff
@@ -18,10 +18,10 @@ def _get_extras_reqs_from(name):
 
 
 def _get_all_extras():
-    return set(
+    return {
```
Comment (auvipy) [inline t/distro/test_CI_reqs.py]: is {} OK instead of set() ?

File: t/benchmarks/bench_worker.py [hunk 9]
```diff
@@ -97,7 +97,7 @@ def bench_both(n=DEFAULT_ITS):
 def main(argv=sys.argv):
     n = DEFAULT_ITS
     if len(argv) < 2:
-        print('Usage: {0} [apply|work|both] [n=20k]'.format(
+        print('Usage: {} [apply|work|both] [n=20k]'.format(
```
Comment (auvipy) [inline t/benchmarks/bench_worker.py]: f ' '

File: t/distro/test_CI_reqs.py (hunk 8 above)
Comment (georgepsarakis) [reply t/distro/test_CI_reqs.py]: This is a set comprehension.

**Comment by thedrow** [comment]
@auvipy Should we merge this?
You didn't approve the PR yet.

File: celery/utils/collections.py [hunk 10]
```diff
@@ -132,7 +131,7 @@ def __getattr__(self, k):
             return self[k]
         except KeyError:
             raise AttributeError(
-                '{0!r} object has no attribute {1!r}'.format(
+                '{!r} object has no attribute {!r}'.format(
```
Comment (auvipy) [inline celery/utils/collections.py]: f

File: celery/utils/collections.py [hunk 11]
```diff
@@ -609,7 +608,7 @@ def update(self, other):
                 if not isinstance(inserted, float):
                     raise ValueError(
                         'Expecting float timestamp, got type '
-                        '{0!r} with value: {1}'.format(
+                        '{!r} with value: {}'.format(
```
Comment (auvipy) [inline celery/utils/collections.py]: f

File: celery/utils/collections.py [hunk 12]
```diff
@@ -787,7 +786,7 @@ def _pop_to_evict(self):
 
     def __repr__(self):
         # type: () -> str
-        return '<{0}: {1}/{2}>'.format(
+        return '<{}: {}/{}>'.format(
```
Comment (auvipy) [inline celery/utils/collections.py]: f

File: celery/utils/collections.py [hunk 13]
```diff
@@ -917,7 +916,7 @@ def _pop_to_evict(self):
...
```
Comment (auvipy) [inline celery/utils/collections.py]: f

File: celery/utils/debug.py [hunk 14]
```diff
@@ -35,7 +34,7 @@
 def _on_blocking(signum, frame):
     import inspect
     raise RuntimeError(
-        'Blocking detection timed-out at: {0}'.format(
+        'Blocking detection timed-out at: {}'.format(
```
Comment (auvipy) [inline celery/utils/debug.py]: f

File: celery/utils/debug.py [hunk 15]
```diff
@@ -133,7 +132,7 @@ def hfloat(f, p=5):
 def humanbytes(s):
     """Convert bytes to human-readable form (e.g., KB, MB)."""
     return next(
-        '{0}{1}'.format(hfloat(s / div if div else s), unit)
+        '{}{}'.format(hfloat(s / div if div else s), unit)
```
Comment (auvipy) [inline celery/utils/debug.py]: f

File: celery/utils/dispatch/signal.py [hunk 16]
```diff
@@ -356,7 +355,7 @@ def _remove_receiver(self, receiver=None):
 
     def __repr__(self):
         """``repr(signal)``."""
-        return '<{0}: {1} providing_args={2!r}>'.format(
+        return '<{}: {} providing_args={!r}>'.format(
```
Comment (auvipy) [inline celery/utils/dispatch/signal.py]: f

File: celery/utils/dispatch/weakref_backports.py [hunk 17]
```diff
@@ -28,7 +28,7 @@ def __new__(cls, meth, callback=None):
             func = meth.__func__
         except AttributeError:
             raise TypeError(
-                "Argument should be a bound method, not {0}".format(
+                "Argument should be a bound method, not {}".format(
```
Comment (auvipy) [inline celery/utils/dispatch/weakref_backports.py]: f

File: celery/utils/imports.py [hunk 18]
```diff
@@ -161,7 +160,7 @@ def load_extension_classes(namespace):
             cls = symbol_by_name(class_name)
         except (ImportError, SyntaxError) as exc:
             warnings.warn(
-                'Cannot load {0} extension {1!r}: {2!r}'.format(
+                'Cannot load {} extension {!r}: {!r}'.format(
```
Comment (auvipy) [inline celery/utils/imports.py]: f

File: celery/utils/saferepr.py [hunk 19]
```diff
@@ -103,7 +102,7 @@ def _chainlist(it, LIT_LIST_SEP=LIT_LIST_SEP):
 
 def _repr_empty_set(s):
     # type: (Set) -> str
-    return '%s()' % (type(s).__name__,)
+    return '{}()'.format(type(s).__name__)
```
Comment (auvipy) [inline celery/utils/saferepr.py]: f

File: celery/utils/saferepr.py [hunk 20]
```diff
@@ -123,9 +122,9 @@ def _format_binary_bytes(val, maxlen, ellipsis='...'):
     if maxlen and len(val) > maxlen:
         # we don't want to copy all the data, just take what we need.
         chunk = memoryview(val)[:maxlen].tobytes()
-        return _bytes_prefix("'{0}{1}'".format(
+        return _bytes_prefix("'{}{}'".format(
```
Comment (auvipy) [inline celery/utils/saferepr.py]: f

File: celery/utils/saferepr.py [hunk 21]
```diff
@@ -123,9 +122,9 @@ def _format_binary_bytes(val, maxlen, ellipsis='...'):
...
             _repr_binary_bytes(chunk), ellipsis))
-    return _bytes_prefix("'{0}'".format(_repr_binary_bytes(val)))
+    return _bytes_prefix("'{}'".format(_repr_binary_bytes(val)))
```
Comment (auvipy) [inline celery/utils/saferepr.py]: f

File: celery/utils/saferepr.py [hunk 22]
```diff
@@ -153,15 +152,15 @@ def _format_chars(val, maxlen):
     if isinstance(val, bytes):  # pragma: no cover
         return _format_binary_bytes(val, maxlen)
     else:
-        return "'{0}'".format(truncate(val, maxlen).replace("'", "\\'"))
+        return "'{}'".format(truncate(val, maxlen).replace("'", "\\'"))
```
Comment (auvipy) [inline celery/utils/saferepr.py]: f

File: celery/utils/saferepr.py [hunk 23]
```diff
@@ -153,15 +152,15 @@ def _format_chars(val, maxlen):
...
 
 
 def _repr(obj):
     # type: (Any) -> str
     try:
         return repr(obj)
     except Exception as exc:
-        return '<Unrepresentable {0!r}{1:#x}: {2!r} {3!r}>'.format(
+        return '<Unrepresentable {!r}{:#x}: {!r} {!r}>'.format(
```
Comment (auvipy) [inline celery/utils/saferepr.py]: f

File: celery/utils/saferepr.py [hunk 24]
```diff
@@ -198,8 +197,8 @@ def _reprseq(val, lit_start, lit_end, builtin_type, chainer):
     if type(val) is builtin_type:  # noqa
         return lit_start, lit_end, chainer(val)
     return (
-        _literal('%s(%s' % (type(val).__name__, lit_start.value), False, +1),
-        _literal('%s)' % (lit_end.value,), False, -1),
+        _literal('{}({}'.format(type(val).__name__, lit_start.value), False, +1),
```
Comment (auvipy) [inline celery/utils/saferepr.py]: f

File: celery/utils/serialization.py [hunk 25]
```diff
@@ -264,7 +263,7 @@ def jsonify(obj,
     else:
         if unknown_type_filter is None:
             raise ValueError(
-                'Unsupported type: {0!r} {1!r} (parent: {2})'.format(
+                'Unsupported type: {!r} {!r} (parent: {})'.format(
```
Comment (auvipy) [inline celery/utils/serialization.py]: f

File: celery/utils/time.py [hunk 26]
```diff
@@ -72,7 +71,7 @@ def __init__(self):
         tzinfo.__init__(self)
 
     def __repr__(self):
-        return '<LocalTimezone: UTC{0:+03d}>'.format(
+        return '<LocalTimezone: UTC{:+03d}>'.format(
```
Comment (auvipy) [inline celery/utils/time.py]: f

File: celery/backends/amqp.py [hunk 27]
```diff
@@ -27,7 +26,7 @@ def repair_uuid(s):
     # Historically the dashes in UUIDS are removed from AMQ entity names,
     # but there's no known reason to.  Hopefully we'll be able to fix
     # this in v4.0.
-    return '%s-%s-%s-%s-%s' % (s[:8], s[8:12], s[12:16], s[16:20], s[20:])
+    return '{}-{}-{}-{}-{}'.format(s[:8], s[8:12], s[12:16], s[16:20], s[20:])
```
[the remark this answers is not in the document]
Comment (thedrow) [reply celery/backends/amqp.py]: Pretty sure we need to remove this one in 5.0 anyway.

File: celery/bin/amqp.py [hunk 28]
```diff
@@ -104,7 +103,7 @@ def dump_message(message):
 
 
 def format_declare_queue(ret):
-    return 'ok. queue:{0} messages:{1} consumers:{2}.'.format(*ret)
+    return 'ok. queue:{} messages:{} consumers:{}.'.format(*ret)
```
[the remark this answers is not in the document]
Comment (thedrow) [reply celery/bin/amqp.py]: Actually this one is fine.
We're unpacking the array into the string.
I don't think it is possible to do the same with f strings here,

File: celery/bin/amqp.py [hunk 29]
```diff
@@ -217,7 +216,7 @@ def do_exit(self, *args):
 
     def display_command_help(self, cmd, short=False):
         spec = self.amqp[cmd]
-        self.say('{0} {1}'.format(cmd, spec.format_signature()))
+        self.say('{} {}'.format(cmd, spec.format_signature()))
```
[the remark this answers is not in the document]
Comment (thedrow) [reply celery/bin/amqp.py]: Will be refactored in #5718.

File: celery/bin/base.py [hunk 30]
```diff
@@ -365,7 +364,7 @@ def ask(self, q, choices, default=None):
                         for c in choices]
         schoices = '/'.join(schoices)
 
-        p = '{0} ({1})? '.format(q.capitalize(), schoices)
+        p = '{} ({})? '.format(q.capitalize(), schoices)
```
[the remark this answers is not in the document]
Comment (thedrow) [reply celery/bin/base.py]: Will be refactored in #5718.

File: celery/bin/celery.py [hunk 31]
```diff
@@ -428,8 +427,8 @@ def on_usage_error(self, exc, command=None):
             helps = '{self.prog_name} {command} --help'
         else:
             helps = '{self.prog_name} --help'
-        self.error(self.colored.magenta('Error: {0}'.format(exc)))
-        self.error("""Please try '{0}'""".format(helps.format(
+        self.error(self.colored.magenta(f'Error: {exc}'))
+        self.error("""Please try '{}'""".format(helps.format(
```
[the remark this answers is not in the document]
Comment (thedrow) [reply celery/bin/celery.py]: Will be refactored in #5718.

File: celery/bin/celery.py [hunk 32]
```diff
@@ -501,13 +500,13 @@ def get_command_info(cls, command, indent=0,
         colored = term.colored() if colored is None else colored
         colored = colored.names[color] if color else lambda x: x
         obj = cls.commands[command]
-        cmd = 'celery {0}'.format(colored(command))
+        cmd = 'celery {}'.format(colored(command))
```
[the remark this answers is not in the document]
Comment (thedrow) [reply celery/bin/celery.py]: Will be refactored in #5718.

File: celery/bin/celery.py [hunk 33]
```diff
@@ -518,7 +517,7 @@ def list_commands(cls, indent=0, colored=None, app=None):
         ret = []
         for command_cls, commands, color in command_classes:
             ret.extend([
-                text.indent('+ {0}: '.format(white(command_cls)), indent),
+                text.indent('+ {}: '.format(white(command_cls)), indent),
```
[the remark this answers is not in the document]
Comment (thedrow) [reply celery/bin/celery.py]: Will be refactored in #5718.

File: celery/bin/control.py [hunk 34]
```diff
@@ -45,7 +45,7 @@ def get_command_info(cls, command,
         else:
             help = None
         return text.join([
-            '|' + text.indent('{0}{1} {2}'.format(
+            '|' + text.indent('{}{} {}'.format(
```
[the remark this answers is not in the document]
Comment (thedrow) [reply celery/bin/control.py]: Will be refactored in #5718.

File: celery/bin/control.py [hunk 35]
```diff
@@ -62,7 +62,7 @@ def list_commands(cls, indent=0, prefix='',
             for c in sorted(choices))
 
     def usage(self, command):
-        return '%(prog)s {0} [options] {1} <command> [arg1 .. argN]'.format(
+        return '%(prog)s {} [options] {} <command> [arg1 .. argN]'.format(
```
[the remark this answers is not in the document]
Comment (thedrow) [reply celery/bin/control.py]: Will be refactored in #5718.

File: celery/bin/control.py [hunk 36]
```diff
@@ -71,26 +71,26 @@ def call(self, *args, **kwargs):
...
-                'Missing {0.name} method.  See --help'.format(self))
+                f'Missing {self.name} method.  See --help')
         return self.do_call_method(args, **kwargs)
 
     def _ensure_fanout_supported(self):
         with self.app.connection_for_write() as conn:
             if not conn.supports_exchange_type('fanout'):
                 raise self.Error(
-                    'Broadcast not supported by transport {0!r}'.format(
+                    'Broadcast not supported by transport {!r}'.format(
```
[the remark this answers is not in the document]
Comment (thedrow) [reply celery/bin/control.py]: Will be refactored in #5718.

File: celery/bin/control.py [hunk 37]
```diff
@@ -137,7 +137,7 @@ def _consume_args(self, meta, method, args):
                     if meta.variadic:
                         break
                     raise self.Error(
-                        'Command {0!r} takes arguments: {1}'.format(
+                        'Command {!r} takes arguments: {}'.format(
```
[the remark this answers is not in the document]
Comment (thedrow) [reply celery/bin/control.py]: Will be refactored in #5718.

File: celery/bin/control.py [hunk 38]
```diff
@@ -233,5 +233,5 @@ def run(self, *args, **kwargs):
                              status=EX_UNAVAILABLE)
         nodecount = len(replies)
         if not kwargs.get('quiet', False):
-            self.out('\n{0} {1} online.'.format(
+            self.out('\n{} {} online.'.format(
```
[the remark this answers is not in the document]
Comment (thedrow) [reply celery/bin/control.py]: Will be refactored in #5718.

File: celery/bin/graph.py [hunk 39]
```diff
@@ -26,7 +25,7 @@ def run(self, what=None, *args, **kwargs):
         if not what:
             raise self.UsageError('missing type')
         elif what not in map:
-            raise self.Error('no graph {0} in {1}'.format(what, '|'.join(map)))
+            raise self.Error('no graph {} in {}'.format(what, '|'.join(map)))
```
[the remark this answers is not in the document]
Comment (thedrow) [reply celery/bin/graph.py]: Will be refactored in #5718.

File: celery/bin/graph.py [hunk 40]
```diff
@@ -81,8 +80,8 @@ class Thread(Node):
 
             def __init__(self, label, **kwargs):
                 self.real_label = label
-                super(Thread, self).__init__(
-                    label='thr-{0}'.format(next(tids)),
+                super().__init__(
+                    label='thr-{}'.format(next(tids)),
```
[the remark this answers is not in the document]
Comment (thedrow) [reply celery/bin/graph.py]: Will be refactored in #5718.

File: celery/bin/graph.py [hunk 41]
```diff
@@ -149,11 +148,11 @@ def maybe_abbr(l, name, max=Wmax):
             size = len(l)
             abbr = max and size > max
             if 'enumerate' in args:
-                l = ['{0}{1}'.format(name, subscript(i + 1))
+                l = ['{}{}'.format(name, subscript(i + 1))
```
[the remark this answers is not in the document]
Comment (thedrow) [reply celery/bin/graph.py]: Will be refactored in #5718.

File: celery/bin/graph.py [hunk 42]
```diff
@@ -149,11 +148,11 @@ def maybe_abbr(l, name, max=Wmax):
...
                      for i, obj in enumerate(l)]
             if abbr:
                 l = l[0:max - 1] + [l[size - 1]]
-                l[max - 2] = '{0}⎨…{1}⎬'.format(
+                l[max - 2] = '{}⎨…{}⎬'.format(
```
[the remark this answers is not in the document]
Comment (thedrow) [reply celery/bin/graph.py]: Will be refactored in #5718.

File: celery/bin/list.py [hunk 43]
```diff
@@ -34,10 +34,10 @@ def run(self, what=None, *_, **kw):
         available = ', '.join(topics)
         if not what:
             raise self.UsageError(
-                'Missing argument, specify one of: {0}'.format(available))
+                f'Missing argument, specify one of: {available}')
         if what not in topics:
             raise self.UsageError(
-                'unknown topic {0!r} (choose one of: {1})'.format(
+                'unknown topic {!r} (choose one of: {})'.format(
```
[the remark this answers is not in the document]
Comment (thedrow) [reply celery/bin/list.py]: Will be refactored in #5718.

File: celery/bin/logtool.py [hunk 44]
```diff
@@ -36,7 +35,7 @@ class _task_counts(list):
 
     @property
     def format(self):
-        return '\n'.join('{0}: {1}'.format(*i) for i in self)
+        return '\n'.join('{}: {}'.format(*i) for i in self)
```
[the remark this answers is not in the document]
Comment (thedrow) [reply celery/bin/logtool.py]: Will be refactored in #5718.

File: celery/bin/logtool.py [hunk 45]
```diff
@@ -137,7 +136,7 @@ def run(self, what=None, *files, **kwargs):
             raise self.UsageError('missing action')
         elif what not in map:
             raise self.Error(
-                'action {0} not in {1}'.format(what, '|'.join(map)),
+                'action {} not in {}'.format(what, '|'.join(map)),
```
[the remark this answers is not in the document]
Comment (thedrow) [reply celery/bin/logtool.py]: Will be refactored in #5718.

File: celery/bin/multi.py [hunk 46]
```diff
@@ -395,7 +394,7 @@ def on_still_waiting_for(self, nodes):
         num_left = len(nodes)
         if num_left:
             self.note(self.colored.blue(
-                '> Waiting for {0} {1} -> {2}...'.format(
+                '> Waiting for {} {} -> {}...'.format(
```
[the remark this answers is not in the document]
Comment (thedrow) [reply celery/bin/multi.py]: Will be refactored in #5718.

File: celery/bin/worker.py [hunk 47]
```diff
@@ -244,7 +243,7 @@ def run(self, hostname=None, pool_cls=None, app=None, uid=None, gid=None,
             try:
                 loglevel = mlevel(loglevel)
             except KeyError:  # pragma: no cover
-                self.die('Unknown level {0!r}.  Please use one of {1}.'.format(
+                self.die('Unknown level {!r}.  Please use one of {}.'.format(
```
[the remark this answers is not in the document]
Comment (thedrow) [reply celery/bin/worker.py]: Will be refactored in #5718.

File: celery/bootsteps.py [hunk 48]
```diff
@@ -50,7 +49,7 @@ class StepFormatter(GraphFormatter):
     }
 
     def label(self, step):
-        return step and '{0}{1}'.format(
+        return step and '{}{}'.format(
```
[the remark this answers is not in the document]
Comment (thedrow) [reply celery/bootsteps.py]: Too complex to be an f string.
In addition bootsteps will be refactored in Celery 5.x.

File: celery/canvas.py [hunk 49]
```diff
@@ -1376,14 +1374,14 @@ def __repr__(self):
             if isinstance(self.body, _chain):
                 return remove_repeating_from_task(
                     self.body.tasks[0]['task'],
-                    '%({0} | {1!r})'.format(
+                    '%({} | {!r})'.format(
```
[the remark this answers is not in the document]
Comment (thedrow) [reply celery/canvas.py]: Too complex to be an f string.

File: celery/contrib/testing/manager.py [hunk 50]
```diff
@@ -112,20 +112,19 @@ def on_result(task_id, value):
             except (socket.timeout, TimeoutError) as exc:
                 waiting_for = self.missing_results(r)
                 self.remark(
-                    'Still waiting for {0}/{1}: [{2}]: {3!r}'.format(
+                    'Still waiting for {}/{}: [{}]: {!r}'.format(
```
[the remark this answers is not in the document]
Comment (thedrow) [reply celery/contrib/testing/manager.py]: Too complex.

File: celery/events/dumper.py [hunk 51]
```diff
@@ -55,7 +54,7 @@ def on_event(self, ev):
         if type.startswith('task-'):
             uuid = ev.pop('uuid')
             if type in ('task-received', 'task-sent'):
-                task = TASK_NAMES[uuid] = '{0}({1}) args={2} kwargs={3}' \
+                task = TASK_NAMES[uuid] = '{}({}) args={} kwargs={}' \
                     .format(ev.pop('name'), uuid,
```
[the remark this answers is not in the document]
Comment (thedrow) [reply celery/events/dumper.py]: Too complex.

File: celery/utils/dispatch/weakref_backports.py (hunk 17 above)
Comment (thedrow) [reply celery/utils/dispatch/weakref_backports.py]: That's a backport. I don't think we need it.

File: celery/utils/saferepr.py (hunk 22 above)
Comment (thedrow) [reply celery/utils/saferepr.py]: Does not work.
See https://stackoverflow.com/questions/44780357/how-to-use-newline-n-in-f-string-to-format-output-in-python-3-6 for why.

File: celery/utils/dispatch/weakref_backports.py (hunk 17 above)
Comment (auvipy) [reply celery/utils/dispatch/weakref_backports.py]: right

**Comment by auvipy** [comment]
good job. just one suggestion, you could pull from master here, this will keep the v5 PR clean from merge from the master.

**Comment by auvipy** [comment]
t/unit/apps/test_multi.py:8: in <module>

    from celery.apps.multi import (Cluster, MultiParser, NamespacedOptionParser,

E     File "/home/travis/build/celery/celery/celery/apps/multi.py", line 130

E       self.cmd = cmd or f'-m {celery_exe('worker', '--detach')}'

E                                                ^

E   SyntaxError: invalid syntax

**Comment by thedrow** [comment]
> good job. just one suggestion, you could pull from master here, this will keep the v5 PR clean from merge from the master.

Oh I get what you mean now.
I don't think we should rewrite history that way.
You merge when you want to integrate code from one branch to another.
That's part of the history.
PRs are squashed and rebased because that way it makes the history look like one of us committed the change.
The problem with your approach is that the changeset for each PR we have will show code changes which do not necessarily belong in that PR.

We can discuss this elsewhere since this is off-topic.

**Comment by auvipy** [comment]
exec(co, mod.__dict__)

t/unit/bin/test_worker.py:12: in <module>

    from celery.apps import worker as cd

E     File "<fstring>", line 1

E       (' '.join(sys.argv)))

E                           ^

E   SyntaxError: unexpected EOF while parsing

**Pull request merged**